src/main/com/yetanalytics/lrs/pedestal/http/multipart_mixed.cljc

```diff
@@ -18,58 +21,123 @@
-                             body-bytes (.getBytes ^String body-str "UTF-8")]]
-                   {:content-type   (get headers "Content-Type")
-                    :content-length (count body-bytes)
-                    :headers        headers
-                    :input-stream   (ByteArrayInputStream. body-bytes)}))))
+       (with-open [^Scanner scanner (.useDelimiter
+                                     (Scanner. in)
+                                     (re-pattern
+                                      (str (boundary-pat-mid boundary)
+                                           "|"
+                                           (boundary-pat-close boundary))))]
+         ;; Skip the (anchored) opening boundary or throw
+         (.skip scanner
+                (re-pattern
+                 (boundary-pat-open boundary)))
+         (loop [multiparts []]
+           (if (.hasNext scanner)
+             (recur (conj multiparts (parse-part (.next scanner) boundary)))
+             (do
+               ;; Skip the (anchored) close boundary or throw
+               (.skip scanner
+                      (re-pattern
+                       (boundary-pat-close boundary)))
+               ;; return multiparts
+               multiparts))))
        :cljs
-       (let [boundary-pattern (make-boundary-pattern boundary)
-             chunks           (cs/split (cs/trim in) boundary-pattern)]
-         (assert (= "" (first chunks)))
-         (into []
-               (for [file-chunk (rest chunks)
-                     ;; Using cs/split in cljs leaves extra crlfs around content
-                     ;; so we remove them with trim
-                     :let [file-chunk (cs/trim file-chunk)
-                           [headers-str
-                            body-str] (cs/split file-chunk #"\r\n\r\n")
-                           headers    (parse-body-headers headers-str)]]
-                 {:content-type   (get headers "Content-Type")
-                  :content-length (.-length body-str)
-                  :headers        headers
-                  :input-stream   body-str}))))
-    #?@(:clj [(catch AssertionError _
-                (throw (ex-info "Invalid Multipart Body"
-                                {:type ::invalid-multipart-body})))
-              (catch Exception _
-                (throw (ex-info "Incomplete Multipart Request"
-                                {:type ::incomplete-multipart})))]
-        :cljs [(catch js/Error jse
-                 (throw (ex-info "Invalid Multipart Body"
-                                 {:type ::invalid-multipart-body}
-                                 jse)))])))
+       (into []
+             (map #(parse-part % boundary)
+                  (split-multiparts boundary in))))
+    (catch #?(:clj Exception
+              :cljs js/Error) ex
+      (throw (ex-info "Invalid Multipart Body"
+                      {:type ::invalid-multipart-body}
+                      ex)))))
```

Comment:
jinx! I don't re-throw the original because I didn't want to actually change the output, there is handling that only looks for the `::invalid-multipart-body` there. The lower-level error is provided for debugging or later work to make more specific error messages.

Reply:
For instance, when I get a test error in the repl I can walk the cause

Reply:
Backwards compatibility is as good of a reason as any.